src/sankoff_handler.cpp

```diff
@@ -0,0 +1,105 @@
+    // generating p_partials (right and left)
+    std::vector<Eigen::Matrix<double, Eigen::Dynamic, state_count>> p_partials_left(site_pattern_.PatternCount());
+    auto p_partials_right = BlankPartials(topology->Id()+1);
+    for (size_t pattern_idx = 0; pattern_idx < site_pattern_.PatternCount(); pattern_idx++) {
+        p_partials_left[pattern_idx] = GenerateLeavesPartials(topology->Id()+1, pattern_idx);
+    }
+    topology->Postorder([&p_partials_left, &p_partials_right, this](const Node* node) {
+        if (!node->IsLeaf()) {
+            for (size_t pattern_idx = 0; pattern_idx < site_pattern_.PatternCount(); pattern_idx++) {
+                Assert(node->Children().size() == child_count, "Error in SankoffHandler::RunSankoff: Tree should be bifurcating.");
+                // Which child partial is in right or left doesn't actually matter because they are summed when calculating q_partials. In DAG case, 
+                // will want to ensure that left children partials are in p_partials_left
+                p_partials_left[pattern_idx].block<1, state_count>(node->Id(), 0) = ParentPartial(TotalPPartial(p_partials_left[pattern_idx], p_partials_right[pattern_idx], node->Children()[0]->Id()));
+                p_partials_right[pattern_idx].block<1, state_count>(node->Id(), 0) = ParentPartial(TotalPPartial(p_partials_left[pattern_idx], p_partials_right[pattern_idx], node->Children()[1]->Id()));
+            }
+        }
+    });
+
+    // generating q-partials
+    auto q_partials = BlankPartials(topology->Id()+1);
+    topology->Preorder([&q_partials, &p_partials_left, &p_partials_right, this](const Node* node) {
+        if (!node->IsLeaf()) {
+            Assert(node->Children().size() == child_count, "Error in SankoffHandler::RunSankoff: Tree should be bifurcating.");
+            for (size_t pattern_idx = 0; pattern_idx < site_pattern_.PatternCount(); pattern_idx++) {
+                auto partials_from_parent = ParentPartial(q_partials[pattern_idx].block<1, state_count>(node->Id(), 0));
+                for (size_t child = 0; child < child_count; child++) {
+                    size_t sister_idx = child == 0 ? 1 : 0;
+                    auto partials_from_sister = ParentPartial(TotalPPartial(p_partials_left[pattern_idx], p_partials_right[pattern_idx], node->Children()[sister_idx]->Id()));
+                    q_partials[pattern_idx].block<1, state_count>(node->Children()[child]->Id(), 0) = partials_from_sister + partials_from_parent;
+                }
+            }
+        } 
+    });
+
+    return {p_partials_left, p_partials_right, q_partials};
+} 
+
+double SankoffHandler::ParsimonyScore(std::tuple<std::vector<Eigen::Matrix<double, Eigen::Dynamic, SankoffHandler::state_count>>, std::vector<Eigen::Matrix<double, Eigen::Dynamic, SankoffHandler::state_count>>, std::vector<Eigen::Matrix<double, Eigen::Dynamic, SankoffHandler::state_count>>> all_partials, size_t node_id) {

```

Comment:
Why do a tuple here rather than just pass the three named things? 

Would a typedef clean this up?

Are these constant objects? Could we pass references?